Sources/Commands/SwiftPackageTool.swift

```diff
@@ -455,8 +468,38 @@ public enum PackageMode: String, StringEnumArgument {
     case update
     case version
     case help
+
+    // PackageMode is not used as an argument; completions will be
+    // provided by the subparsers.
+    public static var completion: ShellCompletion = .none
 }
 
-extension InitMode: StringEnumArgument {}
-extension ShowDependenciesMode: StringEnumArgument {}
-extension DescribeMode: StringEnumArgument {}
+extension InitMode: StringEnumArgument {
```

Comment:
Ok I've dropped the script generation commands from this PR. I've included information in the generated output on how to use the shell scripts.